Add Index to streams

SteveMicroNova opened the pull request:
Streams were missing the optional value `index`, which is used by RCA streams to show what DAC they use

SteveMicroNova committed:
Add missing Stream attributes from AmpliPi's API

linknum23 commented:
Is there anything else that we could need from models.Stream? Below is the full definition from amplipi

```python
class Stream(Base):
  """ Digital stream such as Pandora, AirPlay or Spotify """
  type: str = Field(description="""stream type

  * pandora
  * airplay
  * dlna
  * internetradio
  * spotify
  * plexamp
  * aux
  * file
  * fmradio
  * lms
  * bluetooth
  * rca
  """)
  # TODO: how to support different stream types
  user: Optional[str] = Field(description='User login')
  password: Optional[str] = Field(description='Password')
  station: Optional[str] = Field(description='Radio station identifier')
  url: Optional[str] = Field(description='Stream url, used for internetradio and file')
  logo: Optional[str] = Field(description='Icon/Logo url, used for internetradio')
  freq: Optional[str] = Field(description='FM Frequency (MHz), used for fmradio')
  client_id: Optional[str] = Field(description='Plexamp client_id, becomes "identifier" in server.json')
  token: Optional[str] = Field(description='Plexamp token for server.json')
  server: Optional[str] = Field(description='Server url')
  index: Optional[int] = Field(description='RCA index')
  disabled: Optional[bool] = Field(
    description="Soft disable use of this stream. It won't be shown as a selectable option")
  ap2: Optional[bool] = Field(description='Is Airplay stream AirPlay2?')
  port: Optional[int] = Field(description='Port used by LMS server for metadata listening')
  browsable: Optional[bool] = Field(description='Can this stream be browsed?')
  temporary: Optional[bool] = Field(description='Will this stream be removed once it is fully disconnected from all sources?')
  has_pause: Optional[bool] = Field(description='This stream can be paused, only used on FilePlayers')
  ```

SteveMicroNova commented:
> Is there anything else that we could need from models.Stream? Below is the full definition from amplipi

I added the one I ran into and needed, I've now added all the others
Not certain we _need_ them, but no reason not to add them while we're doing this

linknum23 commented on pyamplipi/models.py:
```diff
@@ -117,6 +117,14 @@ class Stream(BaseModel):
+    server: Optional[str] = None
+    index: Optional[int] = None
+    disabled: Optional[bool] = False
+    ap2: Optional[bool] = False
```
I would argue that ap2 should be set to None just to avoid inserting this field in post parsing and confusingly adding a field on a stream that is not airplay. This is probably an error in our spec.
